site/src/api/queries/templates.ts

```diff
@@ -278,12 +283,16 @@ export const previousTemplateVersion = (
   };
 };
 
-const waitBuildToBeFinished = async (version: TemplateVersion) => {
+const waitBuildToBeFinished = async (
+  version: TemplateVersion,
+  onRequest?: (data: TemplateVersion) => void,
+) => {
   let data: TemplateVersion;
   let jobStatus: ProvisionerJobStatus;
   do {
     await delay(1000);
     data = await API.getTemplateVersion(version.id);
+    onRequest?.(data);
```

Comment:
It is just a hacky thing but I really think we can improve this by having a watching endpoint in the API side: https://github.com/coder/coder/issues/12274

Reply:
ha, you even used that syntax here!